frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx

```diff
@@ -0,0 +1,337 @@
+  resourceObjPath,
+  HandlePromiseProps,
+  withHandlePromise,
+  convertToBaseValue,
+  humanizeBinaryBytes,
+} from '@console/internal/components/utils';
+import {
+  K8sKind,
+  referenceForModel,
+  k8sCreate,
+  referenceFor,
+  VolumeSnapshotClassKind,
+  StorageClassResourceKind,
+  PersistentVolumeClaimKind,
+  k8sGet,
+  VolumeSnapshotKind,
+} from '@console/internal/module/k8s';
+import { connectToPlural } from '@console/internal/kinds';
+import {
+  PersistentVolumeClaimModel,
+  VolumeSnapshotModel,
+  VolumeSnapshotClassModel,
+  StorageClassModel,
+  NamespaceModel,
+  PersistentVolumeModel,
+} from '@console/internal/models';
+import { PVCDropdown } from '@console/internal/components/utils/pvc-dropdown';
+import { apiVersionForModel } from '@console/kubevirt-plugin/integration-tests/tests/utils/selectors';
+import { getName, getNamespace } from '@console/shared';
+import { PVCStatus } from '@console/internal/components/persistent-volume-claim';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+
+import './_create-volume-snapshot.scss';
+
+const LoadingComponent: React.FC = () => (
+  <Grid className="skeleton-box">
+    <GridItem span={12} className="skeleton-activity" />
+    <GridItem span={12} className="skeleton-activity" />
+    <GridItem span={12} className="skeleton-activity" />
+    <GridItem span={12} className="skeleton-activity" />
+    <GridItem span={12} className="skeleton-activity" />
+    <GridItem span={12} className="skeleton-activity" />
+    <GridItem span={12} className="skeleton-activity" />
+    <GridItem span={12} className="skeleton-activity" />
+  </Grid>
+);
+
+const SnapshotClassDropdown: React.FC<SnapshotClassDropdownProps> = (props) => {
+  const { selectedKey, pvcSC } = props;
+  const kind = referenceForModel(VolumeSnapshotClassModel);
+  const resources = [{ kind }];
+  const [scObj, setSCObj] = React.useState<StorageClassResourceKind>(null);
+  const [scError, setError] = React.useState('');
+
+  React.useEffect(() => {
+    k8sGet(StorageClassModel, pvcSC)
+      .then(setSCObj)
+      .catch((error) => setError(error));
+  }, [pvcSC]);
+
+  const filter = (snapshotClass: VolumeSnapshotClassKind) =>
+    !scObj || !scObj.provisioner ? false : scObj?.provisioner.includes(snapshotClass?.driver);
+
+  if (scError) {
+    return (
+      <Alert
+        className="co-alert co-volume-snapshot__alert-body"
+        variant="danger"
+        title="Error fetching info on claim's provisioner"
+        isInline
+      />
+    );
+  }
+
+  return (
+    <ListDropdown
+      {...props}
+      desc="Volume Snapshot Class with same provisioner as claim"
+      dataFilter={filter}
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Select snapshot class"
+      selectedKey={selectedKey}
+    />
+  );
+};
+
+const PVCSummary: React.FC<PVCSummaryProps> = ({ persistentVolumeClaim }) => {
+  const accessModeLabels = Object.freeze({
```

Comment:
can you reuse `https://github.com/openshift/console/blob/master/frontend/public/components/storage/shared.ts#L21` ?